static/elements/chromedash-userlist.html

```diff
@@ -1,107 +1,94 @@
+<link rel="import" type="css" href="../css/elements/chromedash-userlist.css">
 <template>
-  <link rel="stylesheet" href="../css/elements/chromedash-userlist.css">
   <form id="form" name="user_form" method="post" action="{{action}}" onsubmit="return false;">
     <table>
       <tr>
-        <td><input type="email" placeholder="Email address" name="email" id="id_email" required /></td>
-        <td><input type="submit" on-tap="{{ajaxSubmit}}"></td>
-     </tr>
+        <td><input type="email" placeholder="Email address" name="email" id="id_email" required></td>
+        <td><input type="submit" on-tap="ajaxSubmit"></td>
+      </tr>
     </table>
   </form>
-  <hr>
   <ul id="user-list">
-    <template repeat="{{user in users}}">
-      <li><a href="{{action}}/{{user.id}}" on-tap="{{ajaxDelete}}">delete</a>{{user.email}}</li>
+    <template is="dom-repeat" items="{{users}}" as="user">
+      <li>
+        <a href$="{{computeDeleteHref(action, user)}}" on-tap="ajaxDelete">delete</a><span>{{user.email}}</span>
+      </li>
     </template>
   </ul>
 </template>
+</dom-module>
 <script>
-Polymer({
-  created: function() {
-    this.users = this.users || [];
-  },
-  test: function(e, details, sender) {
-    e.preventDefault();
-  },
-  addUser: function(user) {
-    this.users.splice(0, 0, user);
-  },
-  removeUser: function(idx) {
-    return this.users.splice(idx, 1);
-  },
-  ajaxSubmit: function(e, details, sender) {
-    e.preventDefault();
-
-    if (this.$.form.checkValidity()) {
-      // TODO(ericbidelman): move back to this.$.form.email.value when SD
-      // polyfill merges the commit that wraps element.form.
-      var email = this.$.form.querySelector('input[name="email"]').value;
-
-      var formData = new FormData();
-      formData.append('email', email);
-
-      var xhr = document.createElement('core-xhr');
-      xhr.request({url: this.action, method: 'POST', body: formData, callback:
-        function(response, xhr) {
-          if (xhr.status == 201) {
-            this.addUser(JSON.parse(response));
+  Polymer({
+    is: 'chromedash-userlist',
+    properties: {
+      action: {
+        type: String
+      },
+      users: {
+        type: Array,
+        value: function() { return []; }
+      }
+    },
+    computeDeleteHref: function(action, user) {
+      return action + '/' + user.id;
+    },
+    addUser: function(user) {
+      this.splice('users', 0, 0, user);
+    },
+    removeUser: function(idx) {
+      return this.splice('users', idx, 1);
+    },
+    ajaxSubmit: function(e, details) {
+      e.preventDefault();
+      if (this.$.form.checkValidity()) {
+        // TODO(ericbidelman): move back to this.$.form.email.value when SD
+        // polyfill merges the commit that wraps element.form.
+        var email = this.$.form.querySelector('input[name="email"]').value;
+        var formData = new FormData();
+        formData.append('email', email);
+        var xhr = document.createElement('iron-request');
```

Comment:
I'd never seen `<iron-request>` used directly before, only via `<iron-ajax>`. After reading through the `<iron-request>` code, I think it would be clearer as to what's going on if you named the variable for the element `request` instead of `xhr`, and made an additional change that I'll point out a couple of lines down in the `then()`.